client/service/src/client/call_executor.rs

```diff
@@ -352,7 +351,7 @@ mod tests {
 			wasm_runtime_overwrites: None,
 		};
 
-		let client = substrate_test_runtime_client::client::new_with_backend::<
+		let _client = substrate_test_runtime_client::client::new_with_backend::<
```

Comment:
Why do we need this client? It is not used?

Reply:
Originally it was a new Client and new_block, but then I learned that instantiating the test client auto-inserts a block by default.

Needed because check_overwrite uses the `runtime_version` method which gets a block's state to get the runtime version. 

could use sc_block_builder to build a block with just backend but that still requires something that implements the Runtime Api, therefore a Client

Reply:
Can you please add a comment that you only create the client to insert the genesis block

Reply:
Ahh you already added it :)

Reply:
Done :P